src/Data/Text.hs

```diff
@@ -2043,12 +2043,12 @@ stripSuffix p@(Text _arr _off plen) t@(Text arr off len)
 
 -- | Add a list of non-negative numbers.  Errors out on overflow.
 sumP :: String -> [Int] -> Int
-sumP fun = go 0
-  where go !a (x:xs)
-            | ax >= 0   = go ax xs
+sumP fun = L.foldl' add 0
```

Comment:
Let's add a comment that `sumP` uses `foldl'` to fuse with the `L.map` in `concat` because I don't see myself remembering this if I look at just the code in a few months.

Reply:
Added a comment.